### ISSUE TITLE
Detailslist: Screen reader announce read only on header buttons in Scan on in EDGE Narrator

### ISSUE THREAD

**Issue by SainathBommisetty** [microsoft/fluentui#14560]
### Environment Information

- **Package version(s)**: Latest
- **Browser and OS versions**: EDGE + Narrator

### Describe the issue:

In scan mode off screen reader reads as "Button" and the button is actionable and can click for column headers for sorting, but in scan mode the screen reader reads as "Read only" and the button is not actionable and the user will not be able to click on column headers.

### Please provide a reproduction of the issue in a codepen:

https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist

#### Actual behavior:

In scan mode the screen reader reads as "Read only" and the button is not actionable and the user will not be able to click on any column headers. (for sorting etc)

#### Expected behavior:

In Scan on mode screen reader should read all the header column details and columns should be actionable.
